scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/LazyTokenIterator.scala

```diff
@@ -360,7 +360,7 @@ private[parsers] class LazyTokenIterator private (
           case _: RegionIndent | _: RegionIndentEnum => true
           case x: RegionParen => x.canProduceLF
           case _ => false
-        }
+        } && !next.isLeadingInfixOperator
```

Comment:
Was it a bug in Scala 3 implementation?

Reply:
No, for scala3 NL wasn't produced  but it's in a different branch(`if(dialect.allowsSingnificantIndentation) {...}`).